lib/spaceship/portal/provisioning_profile.rb

```diff
@@ -330,42 +320,19 @@ def repair!
       #  the repair method will generate a profile with a new ID
       def update!
         unless certificate_valid?
-          if self.kind_of?Development
+          if self.is_a? Development
             self.certificates = [Spaceship::Certificate::Development.all.first]
-          elsif self.kind_of?InHouse
+          elsif self.is_a? InHouse
             self.certificates = [Spaceship::Certificate::InHouse.all.first]
           else
-            self.certificates = [Spaceship::Certificate::Production.all.first]  
+            self.certificates = [Spaceship::Certificate::Production.all.first]
           end
         end
 
-        def send_update_request(id, name, distribution_method, app_id, certificates, devices)
-          tries ||= 5
-          client.repair_provisioning_profile!(
-            id,
-            name,
-            distribution_method,
-            app_id,
-            certificates,
-            devices
-          )
-        rescue => ex 
-          unless (tries -= 1).zero?
-            sleep 3
-            retry
-          end
-          raise ex # re-raise the exception
+        client.with_retry do
+          client.repair_provisioning_profile!(id, name, distribution_method, app.app_id, certificates.map { |c| c.id }, devices.map { |d| d.id })
```

Comment:
Could you explain? If the line is too long, it can be broken up, but we get to reuse `with_retry` and not have to define a method inside of another method.